src/agent/inspectordebugapi.ts

```diff
@@ -312,37 +350,65 @@ export class InspectorDebugApi implements debugapi.DebugApi {
     if (line === 1) {
       column += debugapi.MODULE_WRAP_PREFIX_LENGTH - 1;
     }
-    let condition;
-    if (breakpoint.condition) condition = breakpoint.condition;
-    let res = this.v8Inspector.setBreakpointByUrl(
-        line - 1, matchingScript, undefined, column - 1, condition);
-    if (res.error || !res.response) {
+
+    let result =
+        this.setAndStoreBreakpoint(breakpoint, line, column, matchingScript);
+    if (!result) {
       return utils.setErrorStatusAndCallback(
           cb, breakpoint, StatusMessage.BREAKPOINT_SOURCE_LOCATION,
           utils.messages.V8_BREAKPOINT_ERROR);
     }
+
     this.breakpoints[breakpoint.id as string] = new BreakpointData(
-        res.response.breakpointId, true, breakpoint, ast as estree.Program,
-        compile);
+        result.v8BreakpointId, breakpoint, ast as estree.Program,
+        result.locationStr, compile);
+
     this.numBreakpoints++;
     setImmediate(function() {
       cb(null);
     });  // success.
   }
 
+  private setAndStoreBreakpoint(
+      breakpoint: apiTypes.Breakpoint, line: number, column: number,
+      matchingScript: string):
+      {v8BreakpointId: inspector.Debugger.BreakpointId,
+       locationStr: string}|null {
+    let condition;
+    if (breakpoint.condition) condition = breakpoint.condition;
+    // location Str will be a JSON string of Stackdriver breakpoint location.
+    // It will be used as key at locationmapper to ensure there will be no
+    // duplicate breakpoints at the same location.
+    let locationStr = JSON.stringify(breakpoint.location);
+    let v8BreakpointId;  // v8/inspector breakpoint id
+    if (!this.locationmapper[locationStr]) {
+      // The first time when a breakpoint was set to this location.
+      let res = this.v8Inspector.setBreakpointByUrl(
+          line - 1, matchingScript, undefined, column - 1, condition);
+      if (res.error || !res.response) {
+        // Error case.
+        return null;
+      }
+      v8BreakpointId = res.response.breakpointId;
+      this.locationmapper[locationStr] = [];
+      this.breakpointmapper[v8BreakpointId] = [];
+    } else {
+      // Breakpoint found at this location. Acquire the v8/inspector breakpoint
+      // id.
+      v8BreakpointId = this.breakpoints[this.locationmapper[locationStr][0]].id;
+    }
+
+    // Adding current stackdriver breakpoint id to location mapper and
+    // breakpoint mapper.
+    this.locationmapper[locationStr].push(breakpoint.id as string);
+    this.breakpointmapper[v8BreakpointId].push(breakpoint.id as string);
+
+    return {v8BreakpointId: v8BreakpointId, locationStr: locationStr};
+  }
+
   private onBreakpointHit(
       breakpoint: apiTypes.Breakpoint, callback: (err: Error|null) => void,
       callFrames: inspector.Debugger.CallFrame[]): void {
-    // TODO: Address the situation where `breakpoint.id` is `null`.
-    const bp = this.breakpoints[breakpoint.id as string];
-    if (!bp.active) {
```

Comment:
Where did this block go?

Reply:
Per our discussion, inspector protocol cannot check the v8 breakpoint active status. And theoretically breakpoint will not be inactive as we never disable them. I remove this part of code and remove the active field in BreakpointData.